scripts/build-controller-image.sh

```diff
@@ -88,9 +89,17 @@ if ! is_public_ecr_logged_in; then
   aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
 fi
 
+pushd "$TEST_INFRA_DIR" 1>/dev/null
+  # Get the golang version from build_config.yaml
+  GOLANG_VERSION=$(cat build_config.yaml | yq .go_version)
+  BASE_IMAGE_VERSION=$(cat build_config.yaml | yq .eks_distro_version) 
+popd 1>/dev/null
+
+
 pushd "$ROOT_DIR" 1>/dev/null
   # Get the golang version from the code-generator
   GOLANG_VERSION=${GOLANG_VERSION:-"$(go list -f {{.GoVersion}} -m)"}
+  BASE_IMAGE_VERSION=${BASE_IMAGE_VERSION:-"$(echo 2024-08-13-1723575672.2)"}
 popd 1>/dev/null
```

Comment:
We're doing this twice should we get rid of the ROOT_DIR one?

Reply:
this is a failsafe, just in case the image is not being built with test-infra in the expected path